src/adapters/MeetingsAdapter.js

```diff
@@ -0,0 +1,90 @@
+import {throwError} from 'rxjs';
+
+import WebexAdapter from './WebexAdapter';
+
+/**
+ * This is a base class that defines the interface that maps meetings data.
+ * Developers that want to extend `MeetingsAdapter` must implement all of its methods,
+ * adhering to the exact parameters and structure of the returned objects.
+ */
+export default class MeetingsAdapter extends WebexAdapter {
+  /**
+   * A meeting object that allows users to have a WebRTC meeting.
+   *
+   * @typedef {Object} Meeting
+   * @property {string}  ID The meeting identifier.
+   * @property {string}  meetingTitle The title of the meeting.
+   * @property {Date}    meetingTimeStart The time and date of the start of the meeting.
+   * @property {Date}    meetingTimeEnd The time and date of the end of the meeting.
+   * @property {Boolean} localAudioMuted If the local user has muted their audio.
+   * @property {Boolean} localVideoMuted If the local user has muted their video.
+   * @property {Boolean} meetingJoined If the local user has joined the meeting.
+   */
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID  ID of the meeting to get.
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeeting(ID) {
+    return throwError(new Error('getMeeting(ID) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Returns a promise that resolved with the meeting ID that matches
+   * the requested parameters.
+   *
+   * @param {string} destinationID
+   * @param {string} destinationType
+   * @returns {Promise.<string>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeetingID(destinationID, destinationType) {
+    return Promise.reject(new Error('getMeetingID(destinationID, destinationType) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Sets the local audio muted state of a particular meetingID.
+   * Returns a promise that is resolved when the set completes.
+   * Also emits an update to the getMeeting observable with the state update to `localAudioMuted`.
+   *
+   * @param {Boolean} audioMuted
+   * @param {string} meetingID
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  setLocalAudioMuted(audioMuted, meetingID) {
```

Comment:
Shouldn't this be more like `toggleLocalAudioMute`? Otherwise, how do you unmute?
Also, I would rather this return an observable that completes, for consistency with other asynchronous-type methods.

Reply:
We've discovered that giving users a `toggle` function gets meetings into a state of conflict when using it multiple times in a row before waiting on the result. That is why the meetings plugin only has `muteAudio` and `unmuteAudio` functions requiring the developer to explicitly define what they want the end state to be after the result of the command.

I will change the promises to observables.

Reply:
>  Otherwise, how do you unmute?

`setLocalAudioMuted(false)`. I'm not against having a `setLocalAudioUnmuted` function instead.